router/state_forward.go

```diff
@@ -133,3 +157,51 @@ func (s *state) _forward(p *peer, f *types.Frame) error {
 
 	return nil
 }
+
+// _flood sends a frame to all of our connected peers. This is used for
+// flooding the wakeup broadcast to all of our direct peers.
+// Classic flooding works by sending frames to all other peers.
+// Tree flooding works by only sending frames to peers on the same branch.
+func (s *state) _flood(from *peer, f *types.Frame, floodType FloodType) {
+	floodCandidates := make(map[types.PublicKey]*peer)
+	for _, p := range s._peers {
+		if p == nil || p.proto == nil || !p.started.Load() {
+			continue
+		}
+
+		if p == from || p == s.r.local {
+			continue
+		}
+
+		if floodType == TreeFlood {
+			if coords, err := p._coords(); err == nil {
+				if coords.DistanceTo(s._coords()) != 1 {
+					// This peer is not directly on the same branch.
+					continue
+				}
+			} else {
+				continue
+			}
+		}
+
+		if p, ok := floodCandidates[p.public]; ok {
+			if p.peertype <= from.peertype {
```

Comment:
In addition to this, it might be worth also checking the `receiveOrder` of the last received tree announcement too, since that should find the lowest latency path.

Reply:
This is actually wrong for another reason. It is incorrectly comparing with the `from` peer and not the peer currently being considered to replace the existing candidate.
I'll fix both things at the same time.